offchain-modules/packages/x/src/db/ckb.ts

```diff
@@ -87,4 +98,35 @@ export class CkbDb {
     const dbRecords = records.map((r) => btcUnlockRepo.create(r));
     await btcUnlockRepo.save(dbRecords);
   }
+
+  async removeUnconfirmedCkbBurn(confirmedBlockHeight: number): Promise<DeleteResult> {
+    return this.connection
+      .getRepository(CkbBurn)
+      .createQueryBuilder()
+      .delete()
+      .where('block_number > :blockNumber', { blockNumber: confirmedBlockHeight })
+      .execute();
+  }
+
+  async getUnconfirmedCkbBurnToConfirm(confirmedBlockHeight: number, limit = 100): Promise<CkbBurn[]> {
+    return this.connection
+      .getRepository(CkbBurn)
+      .createQueryBuilder()
+      .select()
+      .where('block_number <= :confirmedHeight And confirm_status = "unconfirmed"', {
+        confirmedHeight: confirmedBlockHeight,
+      })
+      .limit(limit)
+      .getMany();
+  }
+
+  async updateCkbBurnConfirmStatus(txHashes: string[]): Promise<UpdateResult> {
+    return this.connection
+      .getRepository(CkbBurn)
+      .createQueryBuilder()
+      .update()
+      .set({ confirmStatus: 'confirmed' })
+      .where('ckb_tx_hash in (:txHashes)', { txHashes: txHashes.join(',') })
+      .execute();
+  }
```

Comment:
have you test this function?
seems the sql will be 

```
ckb_tx_hash in (xxxx,yyyy,zzzz)
```

but the right one should be

```
ckb_tx_hash in ('xxxx','yyyy','zzzz')
```